Repositories/UserRepository.cs

```diff
@@ -65,11 +67,11 @@ public async Task<User> Update(string id, User data)
         {
             try
             {
-                await GetCollection().UpdateOneAsync<User>(x => x.Id == id.ToString(),
+                await GetCollection().UpdateOneAsync<User>(x => x.Id == id,
                     Builders<User>.Update.Set(x => x.Username, data.Username).Set(x => x.Password, data.Password)
                         .Set(x => x.Email, data.Email).Set(x => x.PhoneNumber, data.PhoneNumber));
-                var userList = await GetCollection().AsQueryable().ToListAsync();
-                return userList.FirstOrDefault(x => x.Id == data.Id);
+                
+                return await GetCollection().AsQueryable().FirstOrDefaultAsync(u => u.Id == id);
```

Comment:
@christopherplummer More querying the database server instead of grabbing the whole collection and doing it in the API.